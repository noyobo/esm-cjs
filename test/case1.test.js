const code = `import {
  __toModule,
  require_es
} from "../../chunk-ODN36SKU.js";

var import_miniapp_ast2vnode = __toModule(require_es());
var sjsModules = {};
var templateMethods = {};
var includeModules = {};
var ast2VNode = new import_miniapp_ast2vnode.default({
  templateMethods,
  sjsModules,
  includeModules
});
function root(data) {
  this.sjsModules = sjsModules;
  let vnode = ast2VNode.makeMix({
    children: [
      {
        type: "node",
        name: "view",
        attributes: { class: "foo" },
        children: [{ exps: ["key"], expression: 'ty.key+("\u7EC4\u4EF6")' }]
      }
    ]
  }, data);
  return vnode;
}
export {
  root as default,
  templateMethods
};`;

const transform = require('../lib/transform');

describe('transform import case1', () => {
  it('should work', () => {
    expect(transform(code)).toMatchSnapshot();
  });
});
