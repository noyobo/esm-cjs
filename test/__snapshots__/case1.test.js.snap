// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform import case1 should work 1`] = `
"const _chunkODN36SKUJs = require('../../chunk-ODN36SKU.js');

const __toModule = _chunkODN36SKUJs.__toModule;
const require_es = _chunkODN36SKUJs.require_es;

var import_miniapp_ast2vnode = __toModule(require_es());

var sjsModules = {};
var templateMethods = {};
var includeModules = {};
var ast2VNode = new import_miniapp_ast2vnode.default({
  templateMethods,
  sjsModules,
  includeModules
});

function root(data) {
  this.sjsModules = sjsModules;
  let vnode = ast2VNode.makeMix({
    children: [{
      type: \\"node\\",
      name: \\"view\\",
      attributes: {
        class: \\"foo\\"
      },
      children: [{
        exps: [\\"key\\"],
        expression: 'ty.key+(\\"组件\\")'
      }]
    }]
  }, data);
  return vnode;
}

exports.default = root;
exports.templateMethods = templateMethods;"
`;
